dftimewolf/cli/recipes/artifact_grep.py

```diff
@@ -0,0 +1,52 @@
+"""Collect artifacts from hosts using GRR.
+
+- Collect a predefined list of artifacts from hosts using GRR
+- Process them locally with grep to extract keywords
+"""
+from __future__ import unicode_literals
+
+_short_description = ('Fetches requested artifacts from a list of GRR hosts '
+                      'and processes them with grep.')
+
+contents = {
+    'name':
+        'artifact_grep',
+    'short_description': _short_description,
+    'modules': [{
+        'name': 'GRRArtifactCollector',
+        'args': {
+            'hosts': '@hosts',
+            'reason': '@reason',
+            'grr_server_url': '@grr_server_url',
+            'grr_username': '@grr_username',
+            'grr_password': '@grr_password',
+            'artifacts': '@artifacts',
+            'extra_artifacts': '@extra_artifacts',
+            'use_tsk': '@use_tsk',
+            'approvers': '@approvers',
+            'verify': '@verify',
+        },
+    }, {
+        'name': 'GrepperSearch',
+        'args': {
+            'keywords': '@keywords',
+        },
+    }],
+}
+
+args = [
+    ('hosts', 'Comma-separated list of hosts to process', None),
+    ('reason', 'Reason for collection', None),
+    ('--artifacts', 'Comma-separated list of artifacts to fetch '
+     '(override default artifacts)', None),
+    ('--extra_artifacts', 'Comma-separated list of artifacts to append '
+     'to the default artifact list', None),
+    ('--use_tsk', 'Use TSK to fetch artifacts', False),
+    ('--approvers', 'Emails for GRR approval request', None),
+    ('--grr_server_url', 'GRR endpoint', 'http://localhost:8000'),
+    ('--verify', 'Whether to verify the GRR TLS certificate', True),
+    ('--grr_username', 'GRR username', 'admin'),
+    ('--grr_password', 'GRR password', 'admin'),
+    ('--keywords', 'Pipe separated list of keywords to search for '
```

Comment:
```suggestion
    ('--keywords', 'Pipe-separated list of keywords to search for '
```

Also are these keywords or regular expressions?

Also, would it make sense to provide an optional parameter to a file containing a list of keywords to use? (maybe in a future PR)

Reply:
Fixed.
Those are keywords only. Good point to have option for regex as well, I had thought about the file containing keywords. I will add them both for the next PR (I've other functionalities in mind to add as well). Thanks for the suggestions.